operators/src/processing/rasterization/mod.rs

```diff
@@ -0,0 +1,980 @@
+    OperatorName, QueryContext, QueryProcessor, RasterOperator, RasterQueryProcessor,
+    RasterResultDescriptor, SingleVectorSource, TypedRasterQueryProcessor,
+    TypedVectorQueryProcessor,
+};
+
+use crate::error;
+use crate::processing::rasterization::GridOrDensity::Grid;
+use crate::util;
+
+use async_trait::async_trait;
+
+use futures::stream::BoxStream;
+use futures::{stream, StreamExt};
+use geoengine_datatypes::collections::GeometryCollection;
+
+use geoengine_datatypes::primitives::{
+    AxisAlignedRectangle, BoundingBox2D, Coordinate2D, Measurement, RasterQueryRectangle,
+    SpatialPartition2D, SpatialPartitioned, SpatialResolution, VectorQueryRectangle,
+};
+use geoengine_datatypes::raster::{
+    EmptyGrid, GeoTransform, Grid2D, GridOrEmpty, GridSize, GridSpaceToLinearSpace, RasterDataType,
+    RasterTile2D, TilingSpecification,
+};
+
+use num_traits::FloatConst;
+use rayon::prelude::*;
+
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+
+use tracing::span;
+use tracing::Level;
+use typetag::serde;
+
+/// An operator that rasterizes vector data
+pub type Rasterization = Operator<RasterizationParams, SingleVectorSource>;
+
+impl OperatorName for Rasterization {
+    const TYPE_NAME: &'static str = "Rasterization";
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum GridSizeMode {
+    Fixed,
+    Relative,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+#[serde(tag = "type")]
+pub enum GridOrDensity {
+    Grid(GridParams),
+    Density(DensityParams),
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Serialize, Deserialize)]
+pub struct DensityParams {
+    radius: f64,
+    stddev: f64,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GridParams {
+    spatial_resolution: SpatialResolution,
+    origin_coordinate: Coordinate2D,
+    grid_size_mode: GridSizeMode,
+}
+
+/// The parameter spec for `Rasterization`
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RasterizationParams {
+    grid_or_density: GridOrDensity,
+}
+
+#[typetag::serde]
+#[async_trait]
+impl RasterOperator for Rasterization {
+    async fn _initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> util::Result<Box<dyn InitializedRasterOperator>> {
+        let vector_source = self.sources.vector.initialize(context).await?;
+        let in_desc = vector_source.result_descriptor();
+
+        let tiling_specification = context.tiling_specification();
+
+        let out_desc = RasterResultDescriptor {
+            spatial_reference: in_desc.spatial_reference,
+            data_type: RasterDataType::F64,
+            measurement: Measurement::default(),
+            bbox: None,
```

Comment:
I guess no? A safe cut-off point would be the query rectangle + one tile height/width in each direction. But because this depends on the concrete query rectangle and query resolution it is not known at this point.
Maybe another option would be to return the whole area of use of the respective projection (if given in the source descriptor)?